packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -610,6 +712,68 @@ export abstract class BaseService extends Resource
     this.node.defaultChild = this.resource;
   }
 
+  /**
+   *   Enable Deployment Alarms which take advantage of arbitrary alarms and configure them after service initialization
+  */
+  public enableDeploymentAlarms(alarmConfig: DeploymentAlarmConfig) {
+    // Throw an error if deployment alarms are already configured
+    if (this.deploymentAlarms) {
+      throw new Error('Deployment alarms are already configured.');
+    }
+    // Throw an error if alarms array is empty
+    if (alarmConfig.alarms.length === 0) {
+      throw new Error('Specify at least one deployment alarm');
+    }
+    this.deploymentAlarms = {
+      alarmNames: alarmConfig.alarms.map(alarm => alarm.alarmName),
+      enable: true,
+      rollback: alarmConfig.behavior !== AlarmBehavior.FAIL_ON_ALARM,
+    };
+  }
+
+  /**
+   *   Disassociate existing deployment alarms
+  */
+  public disableDeploymentAlarms() {
+    if (this.deploymentAlarms) {
```

Comment:
I think when users try to disassociate, we can just set `deployAlarms` to an empty string list, and `enable` to be false as is now.

Reply:
gotcha
so I made alarmNames: [ ] 
keeping enable : false as it is